### ISSUE TITLE
Can't create `wasm` build

### ISSUE THREAD

**Issue by tpiros** [libjxl/libjxl#2513]
**Describe the bug**
Running `BUILD_TARGET=wasm32 ENABLE_WASM_SIMD=1 emconfigure ./ci.sh release` yields an error

```
CMake Error at third_party/CMakeLists.txt:36 (message):
  Highway library (hwy) not found.  Install libhwy-dev or download it to
  third_party/highway from https://github.com/google/highway .  Highway is
  required to build JPEG XL.  You can run /root/libjxl/deps.sh to download
  this dependency.
```

however: `apt install libhwy-dev`

```
libhwy-dev is already the newest version (1.0.3-2).
0 upgraded, 0 newly installed, 0 to remove and 7 not upgraded.
```

**To Reproduce**
follow the `wasm` build instructions

**Expected behavior**
Expected to have the `wasm` build ready

**Screenshots**

**Environment**
 - OS:  Ubuntu 23.04
 - CPU type: aarch64

**Comment by mo271** [comment]
I think you might need a more up-to-date highway. Perhaps running 

```
git submodule update --init --recursive
```
already fixes this problem?

**Comment by tpiros** [comment]
thanks @mo271 I will give this a go!

**Comment by tpiros** [comment]
yes, I have started everything from scratch earlier today. I installed a new Ubuntu VM and installed everything else from git.

Ubuntu 23.04
Node 20.3.0
emscripten 3.1.33

**Comment by mo271** [comment]
Is this happening for 
```
BUILD_TARGET=wasm32 emconfigure ./ci.sh release
```
or 
```
BUILD_TARGET=wasm32 ENABLE_WASM_SIMD=1 emconfigure ./ci.sh release
```
?

**Comment by tpiros** [comment]
A quick update, I have seen [this discussion on SO](https://stackoverflow.com/questions/55601998/sharederror-closure-compiler-failed-rc-1), and I have removed `--closure 1` from `tools/wasm_demo/CMakeLists.txt`.

**Comment by tpiros** [comment]
Regarding your previous question it was for `BUILD_TARGET=wasm32 ENABLE_WASM_SIMD=1 emconfigure ./ci.sh release`. I can try with the other command as well - one moment.

**Comment by tpiros** [comment]
@mo271 Let me know if removing that flag is OK and also what the final outcome should be by running that command (*). As stated earlier both `emconfigure` commands would return the same error (with having `--closure 1` set).

* I'd love to see that added to the docs - i.e. successfully running the command would result in XYZ, and an explanation/usage instructions for those.

**Comment by mo271** [comment]
The result should be following:
```shell
$ ls build-wasm32/tools/wasm_demo/
CMakeFiles/  cmake_install.cmake  CTestTestfile.cmake  jxl_decoder_for_test.js*  jxl_decoder_for_test.wasm*  jxl_decoder_for_test.worker.js  jxl_decoder.js*  jxl_decoder.wasm*  jxl_decoder.worker.js
```
Then you can follow instructions here to use it:
https://github.com/libjxl/libjxl/tree/main/tools/wasm_demo

Regarding the `--closure 1` flag: Let's check what versions exactly this is affecting, because for me and on the CI it works fine with it.

**Comment by tpiros** [comment]
Thanks @mo271 - confirmed I got those assets as well. I still need to test it but at least the files look OK.

As a side note - and I realise that this is a long shot - but why I ended up here is because initially I wanted to build SSIMULACRA v2 (http://github.com/cloudinary/ssimulacra2) and I had a few issues linking `libhwy` correctly with emscripten and @jonsneyers suggested that I try to build JXL for `wasm` first because it has the same dependencies. So I was wondering if you have ever tried to build SSIMULACRA for `wasm` at all?

**Comment by tpiros** [comment]
hi @mo271 - a kind follow up regarding my previous comment :)

**Comment by mo271** [comment]
Thanks for the ping @tpiros 
I have not tried building SSIMULACRA with `wasm`. 
Perhaps it would be useful to describe the issues you are having in more detail.

**Comment by tpiros** [comment]
I'm sorry for the awful delay here @mo271. Is there another place where we could talk (Discord or something else?)

With all transparency the issue could also stem from the fact that I have limited knowledge of the various build systems and while I was successful in compiling SSIMULACRA v1 to `wasm`, I can't seem to be able to do that using v2.

I have gotten as far as running the following command (after doing the `git clone` and just to confirm, I was also able to build SSIMULACRA itself without any issues)

**Comment by tpiros** [comment]
Thanks @antermin. I sent you a msg over there @mo271 🙏
